src/sif/bicyclecost.cc

```diff
@@ -670,6 +672,13 @@ Cost BicycleCost::EdgeCost(const baldr::DirectedEdge* edge,
   // Create a final edge factor based on total stress and the weighted grade penalty for the edge.
   float factor = 1.0f + grade_penalty[edge->weighted_grade()] + total_stress + surface_factor;
 
+  // Apply factor that defines willingness to use track roads. Pay attention that `track` tag is
+  // already used when calculate `roadway_stress` parameter.
+  // TODO - work with `track` tag in one place
+  if (edge->use() == Use::kTrack) {
+    factor *= track_factor_;
```

Comment:
we check for tracks in the code above https://github.com/valhalla/valhalla/blob/master/src/sif/bicyclecost.cc#L617-L618 . I didn't find the right way how to introduce `track_factor_` here and don't change current behavior, so I decided to apply `track_factor_` to the final `factor` value to be consistent with the logic in other costings

Reply:
that makes me think we should just let bicycle costing alone for the moment. what this is saying is that `use_tracks` collides with `use_roads`. in my opinion `use_roads` is probably the most important bicycle factor (other than what bicycle type you pick) in the costing. @dnesbitt61 opinions?

Reply:
okay, let's remove it for now.

Reply:
I would leave bicycle as is and set the default use_tracks factor to whatever leads to no costing change (0.5 as you have done). A user can override the current behavior to favor/avoid tracks more so than currently if they desire but unless they do so, bicycle routes will not be changed.